### ISSUE TITLE
Return different error codes for only warnings

### ISSUE THREAD

**Issue by ST-o-RM** [squizlabs/PHP_CodeSniffer#1376]
I want make my build unstable when phpcs only warnings found. Is it possible to return different exit code for this case? For example:

0: no errors found
1: errors and warnings found
2: only warnings found

**Comment by jrfnl** [comment]
You can change this behaviour yourself by how you run the command in your build process.

```
// Exit 1 on warnings and errors, 0 on clean.
$> phpcs .

// Exit 1 on errors, ignore warnings completely.
$> phpcs -n .

// Exit 1 on warnings, ignore errors completely.
$> phpcs --error-severity=0 .
```
See for more examples and explanation: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#filtering-errors-and-warnings-based-on-severity

**Comment by gsherwood** [comment]
The most you can do is ignore errors or warnings when calculating the exit code.
https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#ignoring-errors-when-generating-the-exit-code
https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#ignoring-warnings-when-generating-the-exit-code

But that's not what really what you want as you'd need to run PHPCS twice to determine if there were any warnings but no errors.

**Comment by orlangur** [comment]
@ST-o-RM why do you need to check exit code and not the generated `phpcs` report itself?

"build unstable" - sorry, I don't understand what do you mean by these.

I would simply check for error markers in your situation.

For instance, if we talk about `txt` report:
```
--------------------------------------------------------------------------------
FOUND 2 ERROR(S) AFFECTING 2 LINE(S)
--------------------------------------------------------------------------------
  1 | ERROR | End of line character is invalid; expected "\n" but found "\r\n"
 13 | ERROR | Expected 1 space after class name; 2 found
--------------------------------------------------------------------------------
```
it is safe to rely on `ERROR` case-sensitive presence in report.

**Comment by gsherwood** [comment]
Yes, using a report would be another way. Using a JOSN report might be best as you can grab the total errors and warnings reported without going through every file in the report. Example is here: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Reporting#printing-a-json-report

PHPCS allows you to produce multiple reports and send each to either screen or individual files, so you could write the main report to a file and send JSON to STDOUT to parse. Or you could write the JSON to a temp file for reading later.

**Comment by ST-o-RM** [comment]
Yes, we tried to use both report format together - full and JSON. Before this set ignore_warnings_on_exit 1 to pass a decision about build status to script-handler JSON report.
Other solution - to use a summary report and find text something like "A TOTAL OF 0 ERRORS AND 1 WARNING WERE FOUND IN 1 FILE" in report by regular expression.
Thanks

**Comment by Techwolf12** [comment]
So there is no plan to implement a different exit code on warnings than on hard errors?

**Comment by gsherwood** [comment]
> So there is no plan to implement a different exit code on warnings than on hard errors?

There are no plans to change the exit codes at this time. They definitely wont be changed in a 3.x version.
